text_generation/causal_lm/cpp/generate_pipeline/generate_pipeline.hpp

```diff
@@ -0,0 +1,453 @@
+// Copyright (C) 2023-2024 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+
+#pragma once
+
+#include <openvino/openvino.hpp>
+#include <openvino/core/any.hpp>
+#include "generation_config.hpp"
+#include <filesystem>
+#include "group_beam_searcher.hpp"
+
+using GenerationResult = std::vector<std::pair<float, std::vector<int64_t>>>;
+using namespace std;
+
+std::pair<ov::Tensor, ov::Tensor> pad_left(ov::Tensor&& input_ids, ov::Tensor&& attention_mask, int64_t pad_token=2) {
```

Comment:
@apaniukov 
should we by default use left padding in tokenizers? If no - why?

Reply:
Tokenizer doesn't support left padding yet, it will be added in 2024.2

Reply:
a left todo there, will remove it as soon as left padding is enabled by default